infra/api/search/main.tf

```diff
@@ -0,0 +1,55 @@
+locals {
+  # Add environment specific tags
+  tags = merge(module.project_config.default_tags, {
+    environment = var.environment_name
+    description = "Application resources created in ${var.environment_name} environment"
+  })
+}
+
+terraform {
+  required_providers {
+    opensearch = {
+      source  = "opensearch-project/opensearch"
+      version = "2.3.1"
+    }
+  }
+}
+
+data "aws_caller_identity" "current" {}
+
+module "project_config" {
+  source = "../../project-config"
+}
+
+data "aws_ssm_parameter" "search_endpoint_arn" {
+  name = "/search/api-${var.environment_name}/endpoint"
+}
+
+provider "opensearch" {
+  url = data.aws_ssm_parameter.search_endpoint_arn.value
+}
+
+resource "opensearch_role" "admin" {
+  role_name   = "admin"
+  description = "admin test"
+
+  cluster_permissions = ["*"]
+
+  index_permissions {
+    index_patterns  = ["*"]
+    allowed_actions = ["*"]
+  }
+
+  tenant_permissions {
+    tenant_patterns = ["*"]
+    allowed_actions = ["*"]
+  }
+}
```

Comment:
this isn't much better than our current situation 😆 me and @chouinar will work on these permissions